pkg/controller/pubsub/pubsub_init.go

```diff
@@ -89,31 +90,45 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 }
 
 func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
-	providers := psClient.Providers()
-	if len(providers) == 0 {
-		log.Info("No pub sub tool is enabled")
-		return reconcile.Result{}, nil
+	log.Info("Reconcile", "request", request, "namespace", request.Namespace, "name", request.Name)
+
+	deleted := false
+	config := &corev1.ConfigMap{}
+	err := r.Get(ctx, request.NamespacedName, config)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		deleted = true
 	}
 
-	if newConnection, ok := providers[request.Name]; ok {
-		config := &corev1.ConfigMap{}
-		err := r.Get(ctx, request.NamespacedName, config)
+	if deleted {
+		err := r.system.CloseConnection(request.Name)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
-
-		if len(config.Data) == 0 {
-			return reconcile.Result{}, fmt.Errorf(fmt.Sprintf("data missing in configmap %s, unable to configure respective pubsub", request.NamespacedName))
+		log.Info("removed connection", "name", request.Name)
+	} else {
+		providers := psClient.Providers()
+		if len(providers) == 0 {
+			log.Info("No pub sub tool is enabled")
+			return reconcile.Result{}, nil
 		}
-		log.Info(fmt.Sprintf("system state before initiating connection %+v", r.system))
-		err = r.system.InitiateConnection(ctx, newConnection, config.Data)
+
 		if err != nil {
 			return reconcile.Result{}, err
 		}
-		log.Info(fmt.Sprintf("Connection established %+v", r.system))
+		if len(config.Data) == 0 {
```

Comment:
How does this error get surfaced to the user? Can we report status on a configmap somehow?

Prometheus metrics may give some signal (e.g. some connection has an error), but won't be able to tell the user which connection.

Reply:
That's a good idea, reporting errors on configmap to surface it to users

Reply:
Only way I can think of reporting anything is through annotation or label. Do you think that's reasonable to implement?

Reply:
LOL hate to be a broken record, but CRDs support defining a `status` field :p (mostly I want to highlight the various friction points for ConfigMap vs. CRD. The more we find, the more I tend to lean toward CRD actually being simpler even if it does add another kind.

IMO I think annotation would be the only other reasonable choice for reporting status on a resource. We'll probably need byPod status reporting at some point, though since this is only for audit (and audit is a singleton pod), less salient for right now but:

* If whatever reporting scheme we use doesn't support by-pod there is probably a breaking change in our future
* 
* By-pod will require a separate podstatus resource so we can avoid write contention and needing to deal with garbage collection (we let owner references take care of GC for us)
* 
* There is a model for status controllers writing to `status`, not annotations (I don't think it's actually that big a difference, but it is a difference)

Reply:
IMO we've also had things go into alpha without `status` reporting (template expansion is one example, IIRC), so I'm fine if this is categorized as "future work"

Reply:
Cool. we can just log errors for now, and as a follow-up with other design considerations we can either continue down on the configMap path or we can change to CRD.